libnetwork/pasta/pasta_linux.go

```diff
@@ -37,21 +44,25 @@ type SetupOptions struct {
 	ExtraOptions []string
 }
 
-// Setup start the pasta process for the given netns.
-// The pasta binary is looked up in the HelperBinariesDir and $PATH.
-// Note that there is no need any special cleanup logic, the pasta process will
-// automatically exit when the netns path is deleted.
 func Setup(opts *SetupOptions) error {
+	_, err := Setup2(opts)
+	return err
+}
+
+// Setup2 start the pasta process for the given netns.
+// The pasta binary is looked up in the HelperBinariesDir and $PATH.
+// Note that there is no need for any special cleanup logic, the pasta
+// process will automatically exit when the netns path is deleted.
+func Setup2(opts *SetupOptions) (*SetupResult, error) {
```

Comment:
Maybe rename to `SetupWithResult` or `NewSetup` or something that makes it more clear this is the better, generally preferred option?

Reply:
Once I switched the callers in podman and buildah my plan is to rename Setup2 back to Setup()